roachtest: Add mixed version kvprober test

noahstho opened the pull request:
Since the kv prober is a key tool for alerting on issues in production and triaging incidents, we would like to add some additional tests to ensure that it continues to work. This is a simple nightly roach test that ensures we can query crdb_internal.probe_ranges and it successfully returns with no errors.

resolves: #102034
Epic: none
Release note: none

noahstho committed:
roachtest: Add mixed version kvprober test

Since the kv prober is an important tool, we would like to
add some additional tests to ensure that it continues to work.
This is a simple nightly roach test that ensures we can query
crdb_internal.probe_ranges and it successfully returns with no
errors. This PR should be followed by testing the kv_prober_enabled
cluster setting.

resolves: #102034
Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143434)
<!-- Reviewable:end -->

noahstho committed:
Merge branch 'cockroachdb:master' into noahstho/roachtest-kv-prober

noahstho committed:
roachtest: Add mixed version kvprober test
Since the kv prober is used for critical alerts, we would like to
add some additional tests to ensure that it continues to work.
This is a simple nightly roach test that ensures we can query
crdb_internal.probe_ranges and it successfully returns with no
errors.

resolves: #102034
Epic: none
Release note: none
